cli/setup_test.go

```diff
@@ -427,3 +422,76 @@ func TestSetupCLIEmptyStringFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestSetupCLIEmptyServerCert(t *testing.T) {
+	// Test that --server-cert with empty value is valid for on-boarding scenarios
+	// This is the fix for the bug where --server-cert="" was incorrectly rejected
+	flagSet := newFlagSet()
+	ctx, config, runOptions := initCLITest(t, flagSet)
+	defer os.RemoveAll(path.Dir(runOptions.setupOptions.configPath))
+	opts := &runOptions.setupOptions
+
+	// Test explicit empty string --server-cert=""
+	ctx.Set("device-type", "onboard-device")
+	opts.deviceType = "onboard-device"
+	ctx.Set("hosted-mender", "false")
+	opts.hostedMender = false
+	ctx.Set("demo-server", "false")
+	opts.demoServer = false
+	ctx.Set("demo-polling", "false")
+	opts.demoIntervals = false
+	ctx.Set("server-url", "https://onboard.mender.io")
+	opts.serverURL = "https://onboard.mender.io"
+	ctx.Set("server-cert", "")
+	opts.serverCert = ""
+	ctx.Set("update-poll", "100")
+	opts.updatePollInterval = 100
+	ctx.Set("inventory-poll", "200")
+	opts.invPollInterval = 200
+	ctx.Set("retry-poll", "300")
```

Comment:
so why those go into the ctx?

Reply:
I don't understand your question. These are typical configuration parameters.